### DISCUSSION TITLE
custom action for a resource row / bulk?

### DISCUSSION THREAD

**Discussion by wivaku** [filamentphp/filament#6060] [Help] [answered]
I would like to create to create a custom action for a resource.
E.g. "update user to VIP".

I tried:
```
    // UserResource

    public static function table(Table $table): Table
    {
        return $table
            // ...
            ->actions([
                Tables\Actions\Action::make('make VIP')
                    ->action(fn(User $record) => $this->makeVip($record))
                    // ...

    public function makeVip(User $record)
    {
       //
    }
```

But that results in `Using $this when not in object context`

**Comment by danharrin** [comment, marked as answer]
needs to be a static method, or extract into an invokable class or a method on the model

**Reply by tauseedzaman** [reply]
![image](https://github.com/filamentphp/filament/assets/64689921/9d58979c-7484-4d94-a550-19c190232382)
this way worked for me

**Comment by wivaku** [comment]
Perfect. Thanks so much for the super fast and helpful answer (and... fantastic product)!
